src/test/java/com/joss/conductor/mobile/ConductorConfigTest.java

```diff
@@ -0,0 +1,175 @@
+        ConductorConfig config = new ConductorConfig(is);
+
+        Assertions.assertThat(config.getPlatformName())
+                .isEqualByComparingTo(Platform.IOS);
+        Assertions.assertThat(config.getCurrentSchemes())
+                .isEqualTo(new String[] { "longer_timeouts", "ios_device" });
+        Assertions.assertThat(config.getRetries())
+                .isEqualTo(3);
+        Assertions.assertThat(config.getTimeout())
+                .isEqualTo(20);
+        Assertions.assertThat(config.getAppFile())
+                .isEqualTo("./apps/ios.ipa");
+    }
+
+    @Test
+    public void config_overrides_schemes_in_order() {
+        InputStream is = this.getClass().getResourceAsStream("/test_yaml/override_schemes.yaml");
+        ConductorConfig config = new ConductorConfig(is);
+
+        Assertions.assertThat(config.getPlatformName())
+                .isEqualByComparingTo(Platform.IOS);
+        Assertions.assertThat(config.getCurrentSchemes())
+                .isEqualTo(new String[] { "longer_timeouts", "ios_saucelabs" });
+        Assertions.assertThat(config.getRetries())
+                .isEqualTo(1);
+        Assertions.assertThat(config.getTimeout())
+                .isEqualTo(20);
+        Assertions.assertThat(config.getAppFile())
+                .isEqualTo("sauce-storage:mock.zip");
+    }
+
+    @Test
+    public void environment_platform_name_overrides_config() {
+        System.setProperty("conductorPlatformName", "IOS");
+        InputStream is = this.getClass().getResourceAsStream("/test_yaml/all_platforms.yaml");
+        ConductorConfig config = new ConductorConfig(is);
+
+        Assertions.assertThat(config.getPlatformName())
+                .isEqualByComparingTo(Platform.IOS);
+        Assertions.assertThat(config.getAppFile())
+                .isEqualTo("./apps/ios.app");
+    }
+
+    @Test
+    public void environment_schemes_overrides_config() {
+        System.setProperty("conductorCurrentSchemes", "shorter_timeouts,android_device");
+        InputStream is = this.getClass().getResourceAsStream("/test_yaml/all_platforms.yaml");
+        ConductorConfig config = new ConductorConfig(is);
+
+        Assertions.assertThat(config.getPlatformName())
+                .isEqualByComparingTo(Platform.ANDROID);
+        Assertions.assertThat(config.getRetries())
+                .isEqualTo(8);
+        Assertions.assertThat(config.getTimeout())
+                .isEqualTo(5);
+        Assertions.assertThat(config.getUdid())
+                .isEqualTo("auto");
+    }
+
+    @Test
+    public void full_app_path_expands_relative() {
+        InputStream is = this.getClass().getResourceAsStream("/test_yaml/all_platforms.yaml");
+        ConductorConfig config = new ConductorConfig(is);
+
+        Assertions.assertThat(config.getFullAppPath())
+                .isEqualTo(System.getProperty("user.dir") + "/apps/android.apk");
+    }
+
+    @Test
+    public void with_scheme_does_not_resolve() {
```

Comment:
I don't think this test name reflects the actual test: Isn't it verifying the hard coded `sauce-storage` app path does no get extended with the user dir? I would suggest renaming to something like: `app_path_sauce_storage`

Reply:
True - I was originally hoping that this would test urls as well. I'll adjust this and add a new test.